truex-ios-mobile-reference-app/UnlockSoloViewController.m

```diff
@@ -0,0 +1,313 @@
+//
+//  UnlockSoloViewController.m
+//  truex-ios-mobile-reference-app
+//
+//  Created by kyle on 9/27/21.
+//  Copyright © 2021 true[X]. All rights reserved.
+//
+
+#import "UnlockSoloViewController.h"
+#import "WebViewViewController.h"
+#import <TruexAdRenderer/TruexAdRenderer.h>
+
+@interface UnlockSoloViewController ()
+
+@property TruexAdRenderer* activeAdRenderer;
+@property NSMutableDictionary* vastDictionary;
+
+@property (weak, nonatomic) IBOutlet UISwitch *unlocked;
+
+// internal states
+@property NSMutableDictionary* currentPointer;
```

Comment:
This variable name is a little confusing. Is this a pointer to a UI element?

Reply:
I need to translate the XML into a dictionary, and I need some global variables to help me do it. Basically, I need to remember where I am in the tree-- this variable helps me to it.